### PR TITLE
Implement actual code to send email notifications

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR implements email sending code based on standard go smtp library reference.

## Related issue number (if any)
ENG-2268

## Loom demo (if any)
First part of https://www.loom.com/share/b7929818b83743d881db8d712675fdc3

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/notification/email.go [hunk 1]
```diff
@@ -27,9 +29,30 @@ func (e *EmailNotification) Level() shared.NotificationLevel {
...
-	// TODO: Implement
-	return nil
+	auth := smtp.PlainAuth(
+		"", // identity
+		e.conf.User,
+		e.conf.Password,
+		e.conf.Host,
+	)
+
+	fullMsg := fullMessage("aqueduct notification", e.conf.User, e.conf.Targets, msg)
```
Comment (agiron123) [inline src/golang/lib/notification/email.go:48]: Should we change the subject based on the result of the notification?

Or is that out of scope here?

**Review by agiron123 — APPROVED** [review]
Code here looks good. Left a small comment about the subject text.

File: src/golang/lib/notification/email.go (hunk 1 above)
Comment (likawind) [reply src/golang/lib/notification/email.go:48]: Yeah we do plan to make it more dynamic. https://linear.app/aqueducthq/issue/ENG-2367/[m4]-add-proper-subject-for-email-notifications

I'd like to do it in a separate PR but it's def. something we want for release

**Comment by likawind** [comment]
@cw75 @hsubbaraj-spiral I'm going to merge this PR now to unblock the whole stack. But feel free to drop additional comments and I can address separately.

**Pull request merged**